chia/solver/solver_api.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, ClassVar, Optional, cast
+
+from chia.protocols.outbound_message import Message, make_msg
+from chia.protocols.protocol_message_types import ProtocolMessageTypes
+from chia.protocols.solver_protocol import SolverInfo, SolverResponse
+from chia.server.api_protocol import ApiMetadata
+from chia.solver.solver import Solver
+
+
+class SolverAPI:
+    if TYPE_CHECKING:
+        from chia.server.api_protocol import ApiProtocol
+
+        _protocol_check: ClassVar[ApiProtocol] = cast("SolverAPI", None)
+
+    log: logging.Logger
+    solver: Solver
+    metadata: ClassVar[ApiMetadata] = ApiMetadata()
+
+    def __init__(self, solver: Solver) -> None:
+        self.log = logging.getLogger(__name__)
+        self.solver = solver
+
+    def ready(self) -> bool:
+        return self.solver.started
+
+    @metadata.request(peer_required=False, reply_types=[ProtocolMessageTypes.solution_response])
+    async def solve(
+        self,
+        request: SolverInfo,
+    ) -> Optional[Message]:
+        """
+        Solve a V2 plot quality to get the full proof of space.
+        This is called by the farmer when it receives V2 qualities from harvester.
+        """
+        if not self.solver.started:
+            self.log.error("Solver is not started")
+            return None
+
+        self.log.debug(f"Solving quality {request.partial_proof.hex()}with difficulty {request.plot_strength}")
+
+        try:
+            proof = self.solver.solve(request)
+            if proof is None:
+                self.log.warning(f"Solver returned no proof for quality {request.partial_proof.hex()}")
+                return None
+
+            self.log.debug(f"Successfully solved quality, returning {len(proof)} byte proof")
+            return make_msg(
+                ProtocolMessageTypes.solution_response,
+                SolverResponse(proof=proof, partial_proof=request.partial_proof),
```

Comment:
```suggestion
                SolverResponse(proof=proof, partial_proof=partial_proof),
```

Reply:
what do you mean here ? partial_proof is not defined

Reply:
see comment above